pkg/kubelet/config/file_linux.go

```diff
@@ -115,14 +119,20 @@ func (s *sourceFile) processEvent(e *inotify.Event) error {
 			return s.store.Add(pod)
 		}
 	case podDelete:
+		s.locker.Lock()
+		defer s.locker.Unlock()
 		if objKey, keyExist := s.fileKeyMapping[e.Name]; keyExist {
 			pod, podExist, err := s.store.GetByKey(objKey)
 			if err != nil {
 				return err
 			} else if !podExist {
 				return fmt.Errorf("the pod with key %s doesn't exist in cache", objKey)
 			} else {
-				return s.store.Delete(pod)
+				err = s.store.Delete(pod)
+				if err != nil {
```

Comment:
does it make sense to delete it from `fileKeyMapping` if you couldn't delete from `store`?